auto/cflags

```diff
@@ -0,0 +1,29 @@
+
+
+# C compiler flags
+
+
+if [ $NXT_CC_NAME = clang ]; then
+    # Workaround Clang bug
+    nxt_feature="-fdeclspec"
+    nxt_feature_name=
+    nxt_feature_run=
+    nxt_feature_incs="-fdeclspec"
+    nxt_feature_libs=
+    nxt_feature_test="#include <stdlib.h>
+
+                      __declspec(noreturn) static void f(void);
+
+                      static void f(void) {
+                          exit(0);
+                      }
+
+                      int main(void) {
+                          f();
+                      }"
+    . auto/feature
+
+    if [ $nxt_found = yes ]; then
+        NXT_CFLAGS="$NXT_CFLAGS -fdeclspec"
+    fi
+fi
```

Comment:
Do we have a good reason to add it globally rather than only for building Ruby module (and therefore only check it when configuring Ruby module)?

Reply:
No good reason per se.

But are we still in time for appending to the `build/Makefile`'s `CFLAGS` at the time of configuring a module?

Reply:
No good reason per se.

But are we still in time to append to `build/Makefile`'s `CFLAGS` at the time of configuring modules?

Reply:
@alejandro-colomar See `NXT_PERL_CFLAGS` for example. Makefile for Ruby module is generated in the same file where relevant configure tests happen (`auto/modules/ruby`), so it's easy to append additional flags.

Reply:
As I understand, you don't need to append it to `build/Makefile`'s `CFLAGS` for building whole Unit, but only for building the module itself.

Reply:
> As I understand, you don't need to append it to `build/Makefile`'s `CFLAGS` for building whole Unit, but only for building the module itself.

You're right. I don't.

Reply:
Okay, done.  (Only tradeoff/regression wiith respect to the previous implementation (i.e., global `NXT_CFLAGS`), is that now we don't have the test inside a check that the compiler is clang).  But overall, I think it's better: it's restricted to a much smaller part of the build system.

Reply:
I haven't been able to test this yet due to a consistent inability to configure the ruby module (something related to the library path, but let me investigate that tomorrow).

Reply:
>  we don't have the test inside a check that the compiler is clang

Hm.. but why can't we use this check in this case? `configure` for any module can't be called before `configure` for the whole Unit and `NXT_CC_NAME` is saved into the state file (see `auto/save`) and then loaded with `. $NXT_AUTOCONF_DATA`.

Reply:
Didn't know about `NXT_AUTOCONF_DATA`.  Thanks.

Reply:
BTW, I found that the problem to configure is only with Clang... again.

The issue this time is that it doesn't seem to understand `__attribute__((__error__ ("msg")))`.  Will probably open an issue with it.